packages/strapi-utils/lib/content-types.js

```diff
@@ -92,7 +92,11 @@ const getPrivateAttributes = (model = {}) => {
 };
 
 const isPrivateAttribute = (model = {}, attributeName) => {
-  return model && model.privateAttributes && model.privateAttributes.includes(attributeName);
+  if (model && model.attributes && model.attributes[attributeName] && model.attributes[attributeName].private) {
```

Comment:
model.privateAttributes is built at startup and should already contain the private properties. If it is not the case it's not here that we should fix it. but make sure that model.privateAttributes is generated correctly. I think we already have a fix for this in the features/i18n branch we will be merging soon.

Reply:
@alexandrebodin the thing is that privateAttributes is not updated when you select an attribute as private in the content type builder UI. If you set this array with
```json
    "privateAttributes": [
      "created_by",
      "updated_by",
      "created_at",
      "updated_at"
    ]
```
nothing changes only the properties marked as private have a private attribute set to true and this is not being taken into account in the isPrivateAttribute function which only checks the privateAttributes array.

Reply:
As mentionned in my previous comment, the `privateAttributes` property in the code  is defined here https://github.com/strapi/strapi/blob/master/packages/strapi-connector-bookshelf/lib/mount-models.js#L654 on the model.
It is not the value set in the file.

Reply:
Yeah I see it now. thanks for the clarification.

Reply:
fixed